state/factory/erigonstore/contract_storage_namespaced.go

```diff
@@ -0,0 +1,134 @@
+package erigonstore
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/v2/pkg/log"
+	"github.com/iotexproject/iotex-core/v2/state"
+	"github.com/iotexproject/iotex-core/v2/systemcontracts"
+)
+
+type contractStorageNamespacedWrapper struct {
+	standard ContractStorageStandard
+}
+
+// NewContractStorageNamespacedWrapper creates a new ContractStorage wrapper for namespaced storage
+func NewContractStorageNamespacedWrapper(standard ContractStorageStandard) ContractStorage {
+	return &contractStorageNamespacedWrapper{standard: standard}
+}
+
+func (cs *contractStorageNamespacedWrapper) StoreToContract(ns string, key []byte, backend ContractBackend) error {
+	contract, err := cs.storageContract(ns, key, backend)
+	if err != nil {
+		return err
+	}
+	data, err := cs.standard.Serialize()
+	if err != nil {
+		return errors.Wrap(err, "failed to serialize storage standard")
+	}
+	if err := contract.Put(ns, key, systemcontracts.NamespaceGenericValue{PrimaryData: data}); err != nil {
+		return errors.Wrapf(err, "failed to store storage standard to contract %s", contract.Address().Hex())
+	}
+	log.S().Debugf("Stored storage standard to contract %s with key %x, value: %+v", contract.Address().Hex(), key, cs.standard)
+	return nil
+}
+
+func (cs *contractStorageNamespacedWrapper) LoadFromContract(ns string, key []byte, backend ContractBackend) error {
+	contract, err := cs.storageContract(ns, key, backend)
+	if err != nil {
+		return err
+	}
+	value, err := contract.Get(ns, key)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get storage standard from contract %s with key %x", contract.Address().Hex(), key)
+	}
+	if !value.KeyExists {
+		return errors.Wrapf(state.ErrStateNotExist, "storage standard does not exist in contract %s with key %x", contract.Address().Hex(), key)
+	}
+	if err := cs.standard.Deserialize(value.Value.PrimaryData); err != nil {
+		return errors.Wrap(err, "failed to deserialize storage standard")
+	}
+	log.S().Debugf("Loaded storage standard from contract %s with key %x, value: %+v", contract.Address().Hex(), key, cs.standard)
+	return nil
+}
+
+func (cs *contractStorageNamespacedWrapper) DeleteFromContract(ns string, key []byte, backend ContractBackend) error {
+	contract, err := cs.storageContract(ns, key, backend)
+	if err != nil {
+		return err
+	}
+	if err := contract.Remove(ns, key); err != nil {
+		return errors.Wrapf(err, "failed to delete storage standard from contract %s with key %x", contract.Address().Hex(), key)
+	}
+	log.S().Debugf("Deleted storage standard from contract %s with key %x", contract.Address().Hex(), key)
+	return nil
+}
+
+func (cs *contractStorageNamespacedWrapper) ListFromContract(ns string, backend ContractBackend) ([][]byte, []any, error) {
+	contract, err := cs.storageContract(ns, nil, backend)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
if the second parameter `key` could be nil, could it be deleted?
https://github.com/iotexproject/iotex-core/pull/4713/files#diff-9da80c318e8632df1ada1dfb8591fde58e2a5d552dff50190cb8bae9be185538R105
https://github.com/iotexproject/iotex-core/pull/4713/files#diff-ab7c631037f77fb2370b048394c3e86dcdfa2cfeab04ea6d75a86894591ca510R236
https://github.com/iotexproject/iotex-core/pull/4713/files#diff-f8f5e54d3315e09e8fb916f754667e46a305e9a143b8f70dd7b39110e68590d0R87-R95
https://github.com/iotexproject/iotex-core/pull/4713/files#diff-f8f5e54d3315e09e8fb916f754667e46a305e9a143b8f70dd7b39110e68590d0R144-R151
according to the implementations in the follow up PRs, `key` could be deleted.